version: '3.9'

services:
  app:
    image: portainer/portainer-ce:2.18.4-alpine
    ports:
      # - '8000:8000'
      # - '9000:9000'
      - '9443:9443'
    volumes:
      - $(pwd)/docker.sock:/var/run/docker.sock
      - $(pwd)/lib_data:/data
    networks:
      - portainer-net
    healthcheck:
      test:
        [
          'CMD',
          'wget',
          '--tries=1',
          '--spide',
          '--no-check-certificate',
          '--no-verbose',
          'https://localhost:9443'
        ]
      interval: 30s
      timeout: 15s
      retries: 3
      start_period: 10s
    deploy:
      mode: global
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 5
        window: 10s
      resources:
        limits:
          cpus: '1.5'
          memory: 256M

networks:
  portainer-net:
    name: portainer-net
    ipam:
      driver: default
      config:
        - subnet: 172.18.0.0/28
